### DISCUSSION TITLE
Validation on Select Component

### DISCUSSION THREAD

**Discussion by Keltanis** [Megabit/Blazorise#3599] [Q&A]
GIven the Select Component:
```
<Select TValue="int">
    <SelectItem Value="0">-- Select Something --</SelectItem>
    <SelectItem Value="apples">apples</SelectItem>
    <SelectItem Value="pears">pears</SelectItem>
    <SelectItem Value="oranges">oranges</SelectItem>
</Select>
```
How do I validate so that the selected value is not '-- Select Something --'?

**Comment by stsrki** [comment]
Check the example in the demo https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/ValidationsPage.razor#L63

**Comment by Keltanis** [comment]
Hi, this really helps, but if not using numbers for Values, how do you check for a specific string in the value? Any ideas?
  
  
  

----------------------------------------

**Comment by stsrki** [comment]
Ah, I know what you mean now. You have stumbled upon the limitation in the `select` element within the browser/DOM. Zero or empty value will return the content of an `option` element. I advise defining it as

`<SelectItem Value=""></SelectItem>`

or

`<SelectItem Value="invalid-value">-- Select Something --</SelectItem>`

or similar. The trick is to be able to get the actual value without retrieving the content, or text.

**Comment by Keltanis** [comment]
Thanks. Will do.
  
  
  

----------------------------------------
